optuna/terminator/terminator.py

```diff
@@ -0,0 +1,45 @@
+import abc
+from typing import Optional
+
+from optuna._experimental import experimental_class
+from optuna.study.study import Study
+from optuna.terminator.erroreval import BaseErrorEvaluator
+from optuna.terminator.erroreval import CrossValidationErrorEvaluator
+from optuna.terminator.improvement.evaluator import BaseImprovementEvaluator
+from optuna.terminator.improvement.evaluator import DEFAULT_MIN_N_TRIALS
+from optuna.terminator.improvement.evaluator import RegretBoundEvaluator
+from optuna.trial import TrialState
+
+
+class BaseTerminator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def should_terminate(self, study: Study) -> bool:
+        pass
+
+
+@experimental_class("3.2.0")
+class Terminator(BaseTerminator):
+    def __init__(
+        self,
+        improvement_evaluator: Optional[BaseImprovementEvaluator] = None,
+        error_evaluator: Optional[BaseErrorEvaluator] = None,
+        min_n_trials: int = DEFAULT_MIN_N_TRIALS,
```

Comment:
(optional) This might be more consistent with other parameters in Optuna `n_startup_trials`, `n_ei_candidates`, etc. We could also change the constant's name.
```suggestion
        n_min_trials: int = DEFAULT_MIN_N_TRIALS,
```

Reply:
Hmm, I'd like to adopt `min_n_trials` because this is the minimum number of the number of trials.

Reply:
Got it, let's leave it as is.